src/main/java/mari/moita/com/Exercise4.java

```diff
@@ -0,0 +1,43 @@
+package mari.moita.com;
+
+import java.util.List;
+
+public class Exercise4 {
+
+    public void listContainsManAndWoman(List<Person> people) {
+
+        boolean female = false;
+        boolean male = false;
+
+        for (Person p : people) {
+            if ("male".equals(p.getGender())) {
+                male = true;
+            } else if ("female".equals(p.getGender())) {
+                female = true;
+            }
+        }
+        if (!female || !male) {
+            throw new IllegalArgumentException("List must have, at least, one male and one female.");
+        }
+    }
+
+    public Person[] OldestManAndYoungestWoman(List<Person> people) {
+
+        listContainsManAndWoman(people);
+
+        Person[] agePeople = new Person[2];
+
+        for (Person person : people) {
+            if ("male".equals(person.getGender())) {
+                if (agePeople[0] == null || person.getAge() > agePeople[0].getAge()) {
```

Comment:
indexes 0 & 1 could be constants to make it more clear that 0 index is the position that contains the male object, and 1 the female one, like:

private static final int MALE = 0;
private static final int FEMALE = 0;

- agePeople[MALE]
- agePeople[FEMALE]